### PR TITLE
add test for psetv2 p2sh(p2wsh(p2ms())) tx

### LINKED ISSUE #97: Add tests for wrapped segwit (p2sh nested p2wsh)

### LINKED ISSUE #98: use always script type from the prevout!.script
https://github.com/vulpemventures/liquidjs-lib/blob/master/ts_src/psetv2/pset.ts#L419

when we see the script is P2SH we default to legacy signatures, when actually it can happen it's a wrapped segwit so we need segwit v1

### PR DESCRIPTION
**Pull request by louisinger**
Test pass :heavy_check_mark: 

it closes #97 
it closes #98 
@tiero @altafan please review

### CODE REVIEW

File: test/integration/psetv2.spec.ts [hunk 1]
```diff
@@ -1013,6 +1013,88 @@ describe('liquidjs-lib (transactions with psetv2)', () => {
...
+          partialSig: {
+            pubkey: key.publicKey,
+            signature: bscript.signature.encode(
+              key.sign(pset.getInputPreimage(0, Transaction.SIGHASH_ALL)),
+              Transaction.SIGHASH_ALL,
+            ),
+          },
+        };
+        signer.addSignature(0, partialSig, Pset.ECDSASigValidator(ecc));
+      }
```
Comment (altafan) [inline test/integration/psetv2.spec.ts]: can we reuse the `signTransaction` util function for this?

File: ts_src/psetv2/finalizer.ts [hunk 2]
```diff
@@ -34,8 +34,7 @@ export class Finalizer {
       this.finalizeInput(i);
     });
 
-    pset.sanityCheck;
-
+    pset.sanityCheck();
```
Comment (altafan) [inline ts_src/psetv2/finalizer.ts:37]: nice catch! 👀

File: test/integration/psetv2.spec.ts (hunk 1 above)
Comment (louisinger) [reply test/integration/psetv2.spec.ts]: done

**Comment by louisinger** [comment]
fixed `validatePartialSignature` function -> take `redeemScript` in case of wrapped segwit multisig was failing. Now it uses `witnessScript`

**Review by altafan — APPROVED** [review]

**Pull request merged**